main_test.go

```diff
@@ -91,28 +92,105 @@ func TestValidateVargs(t *testing.T) {
 	assert.Equal(t, "my-other-project", vargs.Project)
 }
 
-/*
-//This works in regular go, but not under go test for some reason
-//plugin.MustParse() panics due to stdin EOF
-func TestStdinInput(t *testing.T) {
-
-	tmpfile, _ := ioutil.TempFile("", "stdintest")
-	tmpfileName := tmpfile.Name()
-	defer os.Remove(tmpfileName)
-	tmpfile.Write([]byte(`{"workspace":{"path":"/dev/null"}}`))
-	tmpfile.Close()
-
-	fakeStdin, err := os.Open(tmpfileName)
-	if err != nil {
-		t.Error(err)
+func TestSetupFile(t *testing.T) {
+	tests := []struct {
+		name string
+
+		givenContents string
+		givenVars     map[string]interface{}
+
+		wantError  bool
+		wantOutput string
+	}{
+		{
+			name: "happy path",
+			givenContents: `app:
+  yes: {{ .Yes }}`,
+			givenVars: map[string]interface{}{
+				"Yes": true,
+			},
+
+			wantError: false,
+			wantOutput: `app:
+  yes: true`,
+		},
+		{
+			name: "vars but no references in template",
+			givenContents: `app:
+  yes: true`,
+			givenVars: map[string]interface{}{
+				"Yes": true,
+			},
+
+			wantError: false,
+			wantOutput: `app:
+  yes: true`,
+		},
+		{
+			name: "no vars but references in template",
+			givenContents: `app:
+  yes: {{ .Yes }}`,
+			givenVars: map[string]interface{}{},
+
+			wantError: false,
```

Comment:
I actually expected this case to be `wantError: true`. I see why it is not since this exits on the length check of TemplateVars before parsing the template, but is this a potential confusion trap where the template isn't getting rendered properly and the error returned is nil?

Reply:
Is that the problem of the plugin or the user config problem?
If we want error, the only solution would be to parse the template no matter what, right?

Reply:
Yeah, I'm going to parse the template no matter what. We should catch this stuff before `gcloud`.

Reply:
I think its a user config problem but the nil error could be misleading perhaps. Its probably OK as the worst case (I assume) would be the end user's deployment fails at some point, they check the relevant config files, and see that the template didn't get rendered.

And I think so unfortunately. Would be ugly :(